ref/rust/src/bech32.rs

```diff
@@ -1,12 +1,24 @@
+//! Encode and decode the Bech32 format, with checksums
+//! 
+//! 
+//! 
 
 use std::fmt;
 use std::error::Error;
-use std::ascii::AsciiExt;
 
 #[derive(Debug)]
 pub struct Bech32 {
-    hrp: String,
-    data: Vec<u8>
+    pub hrp: String,
+    pub data: Vec<u8>
+}
+
+impl Bech32 {
+    pub fn clone(&self) -> Bech32 {
+        Bech32 {
+            hrp: self.hrp.clone(),
+            data: self.data.to_vec()
+        }
+    }
 }
```

Comment:
I need to be more familiar with the various things you can `#derive`!

Added in latest commit.